src/libstrongswan/plugins/wolfssl/wolfssl_common.h

```diff
@@ -47,20 +47,30 @@
 /* PARSE_ERROR is an enum entry in wolfSSL - not used in this plugin */
 #define PARSE_ERROR	WOLFSSL_PARSE_ERROR
 
-/* Remap unused enums from the OpenSSL compatibility layer to avoid conflicts */
-#define ASN1_BOOLEAN         REMAP_ASN1_BOOLEAN
-#define ASN1_OID             REMAP_ASN1_OID
-#define ASN1_INTEGER         REMAP_ASN1_INTEGER
-#define ASN1_BIT_STRING      REMAP_ASN1_BIT_STRING
-#define ASN1_IA5STRING       REMAP_ASN1_IA5STRING
-#define ASN1_OCTET_STRING    REMAP_ASN1_OCTET_STRING
-#define ASN1_UTCTIME         REMAP_ASN1_UTCTIME
-#define ASN1_GENERALIZEDTIME REMAP_ASN1_GENERALIZEDTIME
-
 #ifndef WOLFSSL_USER_SETTINGS
 	#include <wolfssl/options.h>
 #endif
+
+#if defined(ECC_TIMING_RESISTANT) && \
+    (!defined(HAVE_FIPS) || \
+     (defined(HAVE_FIPS_VERSION) && (HAVE_FIPS_VERSION >= 5)))
+    #define USE_RNG_FOR_TIMING_RESISTANCE
+#endif
+
+/* Disable inclusion of the wolfSSL OpenSSL compatibility layer header (if
+   configured) as it is not used by the plugin and causes conflicts */
+#define WOLFSSL_OPENSSL_H_
+
+/* Conditionally include wolfssl/ssl.h to avoid conflicts between the wolfSSL
+   and strongSwan ASN1 values in older wolfSSL versions */
```

Comment:
No, the define goes back as far as the /wolfssl/openssl/ssl.h file does, with the file (and openssl headers) first appearing in wolfssl v3.3.3.  This define should apply to any version of wolfssl, unless we change the format of the include file duplicate-inclusion-protection macros at some point in the future, which I think is unlikely.

Reply:
Ah, I didn't realize this is the include guard of that header. So if this was the same since 3.3.3, why is defining `WOLFSSL_COMMON_INC_SSL_H` necessary in `wolfssl_plugin.c`?

Reply:
Thanks for the explanation. Makes sense.

Reply:
Actually, since `wolfssl/ssl.h` is only used in `wolfssl_plugin.c`, would it make sense to just move the `#include` statement?